### DISCUSSION TITLE
target = "_self" on nav links

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vuejs/vitepress#1015] [Q&A] [answered]
I'd like a link within my nav bar to use `target="_self"` rather than the current assumption of `target="_blank". Is there a configuration variable I can pass to my nav config to allow this to happen?

I believe this is happening because the link is a 3rd-party site, but I'd prefer if I had control over it.

**Comment by xfq** [comment]
See https://github.com/vuejs/vitepress/issues/997

**Reply by brc-dd** [reply]
Its probably different than that one. Nav links use that `VPLink` component and target blank is hardcoded there.

**Reply by xfq** [reply]
Ah, indeed. Thanks.

**Comment by brc-dd** [comment, marked as answer]
You can create a `vite.config.ts` inside docs and add this there for now:

```ts
import type { Plugin } from 'vite';
import { defineConfig } from 'vite';

const NavLinkPatch = (): Plugin => ({
  name: 'override-target-blank',
  enforce: 'pre',
  transform: (code, id) => {
    if (id.endsWith('VPLink.vue')) {
      return code.replace('_blank', '_self');
    }
  },
});

export default defineConfig({
  plugins: [NavLinkPatch()],
});
```

**Reply by CodyBontecou** [reply]
Awesome, that worked. Thank you for the quick reply + help!
